subprojects/core/src/integTest/groovy/org/gradle/api/tasks/MissingTaskDependenciesIntegrationTest.groovy

```diff
@@ -562,14 +566,29 @@ The following types/formats are supported:
         failureCauseContains(cause)
     }
 
-    void assertMissingDependency(String producerTask, String consumerTask, File producedConsumedLocation) {
+    void assertMissingDependency(String producerTask, String consumerTask, File... producedConsumedLocations) {
         expectReindentedValidationMessage()
-        def expectedMessage = implicitDependency {
-            at(producedConsumedLocation)
-            consumer(consumerTask)
-            producer(producerTask)
-            includeLink()
+        if (GradleContextualExecuter.configCache) {
+            // Due to extra parallelism with configuration cache missing dependencies detection mechanism
+            // can report multiple errors instead of just one as is the case without configuration cache.
+            def messageMatchers = producedConsumedLocations.collect { producedConsumedLocation ->
+                def message = implicitDependency {
+                    at(producedConsumedLocation)
+                    consumer(consumerTask)
+                    producer(producerTask)
+                    includeLink()
+                }
+                containsNormalizedString(message)
+            }
+            failure.assertThatAllDescriptions(anyOf(messageMatchers))
```

Comment:
💭 This is the main change: We allow multiple errors, but they should match any location error message:
- normally it's just one location and so one message, but it's reported for multiple tasks. Only one task (the one that was adjusted) has multiple locations